superset/db_engine_specs/trino.py

```diff
@@ -58,7 +58,10 @@ def extra_table_metadata(
         table_name: str,
         schema_name: str | None,
     ) -> dict[str, Any]:
-        metadata = {}
+        metadata: dict[str, Any] = {}
+
+        if not database.has_table_by_name(table_name):
```

Comment:
@Khrol the issue with this method is it only checks physical tables (as opposed to tables and views).

Is there any reason not to implement my previous suggestion and override the `TrinoEngineSpec.get_indexes()` method? In addition to working both for tables and views  itwould actually be more efficient as well as it requires one fewer checks, i.e., asking for forgiveness (via a `try`/`except` block) instead of for permission (via the `has_table_by_name()` method).

Reply:
> Is there any reason not to implement my previous suggestion and override the TrinoEngineSpec.get_indexes() method?

I was a bit lost between `TrinoEngineSpec`, `Database` and `Inspector`. 🤦‍♂️  Some methods are static, others are not... 

I understand what you mean now. I'll update PR.